nucleus/payara-modules/nucleus-microprofile/config-service/src/test/java/fish/payara/nucleus/microprofile/config/source/DirConfigSourceTest.java

```diff
@@ -0,0 +1,239 @@
+        props.put(property,
+            new DirConfigSource.DirProperty(
+                "test", FileTime.from(Instant.now()),
+                Paths.get(testDirectory.toString(), "foo.bar/test.ex/one"),
+                testDirectory));
+        source.setProperties(props);
+        
+        // when & then
+        assertTrue(source.isLongerMatchForPath(property, Paths.get(testDirectory.toString(), "foo.bar/test/ex.one.txt")));
+    }
+    
+    @Test
+    public void testCheckLongestMatchForPath_PropNotPresent() {
+        // given
+        Map<String,DirConfigSource.DirProperty> props = new HashMap<>();
+        // a property with a most specific path
+        String property = "foo.bar.test.ex.one";
+        props.put(property,
+            new DirConfigSource.DirProperty(
+                "test", FileTime.from(Instant.now()),
+                Paths.get(testDirectory.toString(), "foo.bar/test.ex/one"),
+                testDirectory));
+        source.setProperties(props);
+        
+        // when & then
+        assertTrue(source.isLongerMatchForPath("foo.bar.test.ex.two", Paths.get(testDirectory.toString(), "foo.bar/test/ex.two.txt")));
+    }
+    
+    @Test
+    public void testRemovePropertyFromPath() {
+        // given
+        Map<String,DirConfigSource.DirProperty> props = new HashMap<>();
+        // a property with a most specific path
+        String property = "foo.bar.test";
+        props.put(property,
+            new DirConfigSource.DirProperty(
+                "test", FileTime.from(Instant.now()),
+                Paths.get(testDirectory.toString(), "foo/bar/test"),
+                testDirectory));
+        source.setProperties(props);
+        assertEquals("test", source.getValue(property));
+        
+        // when
+        source.removePropertyFromPath(Paths.get(testDirectory.toString(), "foo/bar/test"));
+        // then
+        assertTrue(source.getValue(property) == null);
+        
+    }
+
+    @Test
+    public void testInitializeProperties_SimpleFiles() throws IOException {
+        // given
+        // only the most specific should be picked up (=test3)
+        writeFile(testDirectory, "foo.bar.test", "test");
+        writeFile(testDirectory, "foo.bar/test", "test2");
+        writeFile(testDirectory, "foo/bar/test", "test3");
+        
+        //when
+        source.initializePropertiesFromPath(testDirectory);
+        
+        //then
+        assertEquals("test3", source.getValue("foo.bar.test"));
+    }
+    
+    @Test
+    public void testInitializeProperties_IgnoreHidden() throws IOException {
+        // given
+        // none of these should be picked up (hidden file or dir)
+        writeFile(testDirectory, ".hidden.bar.test", "test");
+        writeFile(testDirectory, ".hidden/bar.test", "test");
+        //when
+        source.initializePropertiesFromPath(testDirectory);
+        //then
+        assertEquals(null, source.getValue("hidden.bar.test"));
```

Comment:
This fails on Windows, because it has different condition for a hidden file. Consider explicit check for dot at start rather than `isHidden`.

Reply:
Done. It *should* work now.